### PR TITLE
Lt 2577 freeze products

### PR DESCRIPTION
**Pull request by gponomarev-lykke**

### CODE REVIEW

**Comment by gponomarev-lykke** [comment]
Validations on product freeze look very weird, I think we can simplify some of them. Also we have to remember about a dependency on Corporate Actions api - can it be dropped or refactored?

I tried to introduce a way to validate a list of constraints in ValidationChainEngine and ProductAddOrUpdateValidation. I hoped to reuse it in validations for FreezeInfo, but since it uses some extra fields for validation I'd have to introduce a Context class for ValidationChainEngine. I'm not sure we want to expand on this idea, so I'd like to hear your opinion.

**Comment by gponomarev-lykke** [comment]
Also I had to rebase this branch on LT-2575-rabbitmq-events to get rabbit-mq related changes

File: src/MarginTrading.AssetService.Services/ClientProfileSettingsService.cs [hunk 1]
```diff
@@ -67,6 +70,10 @@ public async Task UpdateAsync(ClientProfileSettings model, string username, stri
 
             await _auditService.TryAudit(correlationId, username, referenceId, AuditDataType.ClientProfileSettings,
                 model.ToJson(), existing.ToJson());
+
+            await _cqrsMessageSender
```
Comment (tarurar) [inline src/MarginTrading.AssetService.Services/ClientProfileSettingsService.cs:74]: Maybe it would be better to audit with subscriber.

**Review by tarurar — DISMISSED** [review]

File: src/MarginTrading.AssetService.SqlRepositories/Extensions/JsonSerializersExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+
+namespace MarginTrading.AssetService.SqlRepositories.Extensions
+{
+    public static class JsonSerializersExtensions
+    {
+        public static string ToJsonWithStringEnums(this object src, bool ignoreNulls = false)
```
Comment (andrey-anokhin) [inline src/MarginTrading.AssetService.SqlRepositories/Extensions/JsonSerializersExtensions.cs:8]: If I am not wrong, we can do that by adding [JsonConverter(typeof(StringEnumConverter))] to the enum and it was done already for the old FreezeInfoContract for example. By the way, why you decided to create a new classes and enums and do not use the old ones?

File: src/MarginTrading.AssetService/appsettings.test.json [hunk 3]
```diff
@@ -4,6 +4,10 @@
       "ApiKey": "margintrading",
       "ServiceUrl": "http://mdm.mt.svc.cluster.local/"
     },
+    "CorporateActionsService": {
```
Comment (andrey-anokhin) [inline src/MarginTrading.AssetService/appsettings.test.json:7]: This should be removed

**Review by andrey-anokhin — APPROVED** [review]

**Comment by andrey-anokhin** [comment]
@gponomarev-lykke I am merging this PR, because it is required for other tasks and I do not want to increase complexity and conflicts. But all comments should be fixed in the scope of the task https://lykke-snow.atlassian.net/browse/LT-2656 when everything related to Asset Service will be merged to master.

**Pull request merged**